### PR TITLE
Init container changes with improved probe times

### PR DESCRIPTION
**Pull request by Tirumalavasa**
https://github.ibm.com/IBMPrivateCloud/roadmap/issues/58986
https://github.ibm.com/IBMPrivateCloud/roadmap/issues/58040

### CODE REVIEW

File: deploy/olm-catalog/ibm-iam-operator/4.1.0/ibm-iam-operator.v4.1.0.clusterserviceversion.yaml [hunk 1]
```diff
@@ -221,13 +221,15 @@ metadata:
...
-      image: icr.io/cpopen/cpfs/icp-platform-auth@sha256:12715da7cb32bc60b9869856c8430e4673a95d9d883f96de3afde5c2fe579437
+      image: icr.io/cpopen/cpfs/icp-platform-auth:4.1.0
     - name: ICP_IDENTITY_PROVIDER_IMAGE
-      image: icr.io/cpopen/cpfs/icp-identity-provider@sha256:15c8a1c9b2384bfa4cff1ae38bf0261492a4fef4d9a4907de3b0e7411cd3c0dd
+      image: icr.io/cpopen/cpfs/icp-identity-provider:4.1.0
     - name: ICP_IDENTITY_MANAGER_IMAGE
-      image: icr.io/cpopen/cpfs/icp-identity-manager@sha256:191c45c6e69849123c8d5f0f7a789cb522b2fb5c2aab75563f43af78150e030a
+      image: icr.io/cpopen/cpfs/icp-identity-manager:4.1.0
+    - name: IM_INITCONTAINER_IMAGE
+      image: icr.io/cpopen/cpfs/im-initcontainer:4.1.0
```
Comment (rashmi43) [inline deploy/olm-catalog/ibm-iam-operator/4.1.0/ibm-iam-operator.v4.1.0.clusterserviceversion.yaml:232]: will we need to add this to pipeline and offline ?

File: deploy/olm-catalog/ibm-iam-operator/4.1.0/ibm-iam-operator.v4.1.0.clusterserviceversion.yaml [hunk 2]
```diff
@@ -389,6 +391,7 @@ spec:
 
     ## Operator versions
      - 4.1.0
+     - 4.0.0
```
Comment (rashmi43) [inline deploy/olm-catalog/ibm-iam-operator/4.1.0/ibm-iam-operator.v4.1.0.clusterserviceversion.yaml:394]: we have 4.0.1 also but not sure we need to mention it here or not?

File: pkg/controller/authentication/job.go [hunk 3]
```diff
@@ -211,7 +211,7 @@ func buildContainer(jobName string, image string, resources *corev1.ResourceRequ
 		{
 			Name:            jobName,
 			Image:           image,
-			ImagePullPolicy: corev1.PullAlways,
+			ImagePullPolicy: corev1.PullIfNotPresent,
```
Comment (rashmi43) [inline pkg/controller/authentication/job.go:214]: is this the same for all images?

**Review by rashmi43 — APPROVED** [review]

File: deploy/olm-catalog/ibm-iam-operator/4.1.0/ibm-iam-operator.v4.1.0.clusterserviceversion.yaml (hunk 1 above)
Comment (yannizhang2019) [reply deploy/olm-catalog/ibm-iam-operator/4.1.0/ibm-iam-operator.v4.1.0.clusterserviceversion.yaml:232]: why did you change from digest to tag?
When using tag for image, the imagePullPolicy must be `Always`. It's a certification requirement.

Comment (Tirumalavasa) [reply deploy/olm-catalog/ibm-iam-operator/4.1.0/ibm-iam-operator.v4.1.0.clusterserviceversion.yaml:232]: > will we need to add this to pipeline and offline ?

Yes, I have already added this image entry to  CS-cd.json file

Comment (Tirumalavasa) [reply deploy/olm-catalog/ibm-iam-operator/4.1.0/ibm-iam-operator.v4.1.0.clusterserviceversion.yaml:232]: > why did you change from digest to tag? When using tag for image, the imagePullPolicy must be `Always`. It's a certification requirement.

@yannizhang2019 , this was a stale digest, All operator repos will just mention the image tag, during promotion to common service catalog, CICD will replace it with actual digest, see here https://github.ibm.com/IBMPrivateCloud/ibm-common-service-catalog/blob/icr-cd/manifests/ibm-iam-operator/4.1.0/ibm-iam-operator.v4.1.0.clusterserviceversion.yaml#L222-L230
https://github.ibm.com/IBMPrivateCloud/ibm-common-service-catalog/blob/icr-cd/manifests/ibm-iam-operator/4.1.0/ibm-iam-operator.v4.1.0.clusterserviceversion.yaml#L501-L524

File: deploy/olm-catalog/ibm-iam-operator/4.1.0/ibm-iam-operator.v4.1.0.clusterserviceversion.yaml (hunk 2 above)
Comment (Tirumalavasa) [reply deploy/olm-catalog/ibm-iam-operator/4.1.0/ibm-iam-operator.v4.1.0.clusterserviceversion.yaml:394]: this is just for book keeping purpose, not really needed

File: pkg/controller/authentication/job.go (hunk 3 above)
Comment (Tirumalavasa) [reply pkg/controller/authentication/job.go:214]: I have made it for the im-initcontainer image, as this is just plain ubi-minimal image, we don't really need Always pullPolicy

**Review by ujjwalchk-it — APPROVED** [review]

**Review by rashmi43 — APPROVED** [review]

**Pull request merged**